### PR TITLE
fix:[CI-9758]: Multi Stage Docker Support

### PR DESCRIPTION
**Pull request by VikasHaritwal93**
This PR is to support multistage docker file for Log Service. This build creation is tested and evidence can be find here:

https://stage.harness.io/ng/account/wFHXHD0RRQWoO8tIZT5YVw/home/orgs/Harness/projects/RELEASEBUILDS/pipelines/logServiceBuild_Multi_Stage_Test/executions?connectorRef=wingssoftwarebuildtooutility&repoName=build-tools-utility&branch=main&storeType=REMOTE

https://app.harness.io/ng/account/vpCkHKsDSxK9_KYfjCTMKA/all/orgs/default/projects/DevSpaces/pipelines/Build_Services/executions/tQlortsFTb-TQxHEenwdIg/pipeline?connectorRef=harnessplinfra&repoName=harness-pl-infra&branch=master&storeType=REMOTE&stage=FHWJCrtRRTiPI4gARiMjsA&stageExecId=obS2yHM0S_2wgwCttc2OEw&step=xK8t2MBJQD-VXauHTlsWgQ&childStage=

https://app.harness.io/ng/account/vpCkHKsDSxK9_KYfjCTMKA/all/orgs/default/projects/DevSpaces/pipelines/Deploy_Services/executions/HzE8PasWQcm0-qwjfgsiZA/pipeline?connectorRef=harnessplinfra&repoName=harness-pl-infra&branch=master&storeType=REMOTE

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54991)

### CODE REVIEW

File: product/log-service/docker/Dockerfile-log-service-multistage [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+RUN mkdir -p /app
+ARG GCP_KEY
+ENV SERVICE_NAME="log-service"
+COPY . /app
+
+WORKDIR /app
+
+ENV BUILD_PURPOSE=DEVELOPMENT
+ENV PLATFORM="jenkins"
```
Comment (vistaarjuneja) [inline product/log-service/docker/Dockerfile-log-service-multistage:23]: Nit: do we need this?

**Review by smjt-h — DISMISSED** [review]

File: product/log-service/build/feature_build.sh [hunk 2]
```diff
@@ -3,10 +3,10 @@
 # that can be found in the licenses directory at the root of this repository, also available at
 # https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
 
-// Copyright 2020 Harness Inc. All rights reserved.
-// Use of this source code is governed by the PolyForm Free Trial 1.0.0 license
-// that can be found in the licenses directory at the root of this repository, also available at
-// https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
+#// Copyright 2020 Harness Inc. All rights reserved.
```
Comment (vistaarjuneja) [inline product/log-service/build/feature_build.sh:6]: there'a already one copyright above, can we just remove this?

Comment (VikasHaritwal93) [reply product/log-service/build/feature_build.sh:6]: Done

**Review by vistaarjuneja — APPROVED** [review]

File: product/log-service/config/manifest.yaml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-BUILD_SCRIPT_PATH: log-service/build/feature_build.sh
-DOCKER_FILE_PATH: /harness/harness-core/product/log-service/docker/Dockerfile
+BUILD_SCRIPT_PATH: product/log-service/build/feature_build.sh
```
Comment (Vbhatia20) [inline product/log-service/config/manifest.yaml:1]: this is not needed, **BUILD_SCRIPT_PATH**

File: product/log-service/config/manifest.yaml [hunk 4]
```diff
@@ -1,4 +1,4 @@
...
+DOCKER_FILE_PATH: /harness/harness-core/product/log-service/docker/Dockerfile-log-service-multistage
 HELM_CHART_SOURCE_DIRECTORY: harness-core/product/log-service/chart
-VERSION_FILE: harness-core/980-commons/src/main/resources-filtered/versionInfo.yaml
+VERSION_FILE: harness-core/980-commons/src/main/resources-filtered/versionInfo.yaml
```
Comment (Vbhatia20) [inline product/log-service/config/manifest.yaml:4]: VERSION_FILE also can be removed and not needed.

**Review by Vbhatia20 — CHANGES_REQUESTED** [review]

**Pull request merged**